src/main/java/frc/robot/RobotContainer.java

```diff
@@ -104,7 +124,33 @@ private void configureDefaultCommands() {
                         () -> true));
     }
 
-    private void configureButtonBindings() {}
+    private Command updateScoreState(ScoreState newState, Set<Subsystem> scoringRequirements) {
+        return Commands.defer(
+                () ->
+                        Commands.runOnce(() -> currentState = newState)
+                                .andThen(
+                                        Commands.repeatingSequence(
+                                                newState.calculateTargets(),
+                                                newState.prepareSubsystems())),
+                scoringRequirements);
+    }
+
+    private void configureButtonBindings() {
+        xboxController.y().whileTrue(commandGroups.intake());
+        xboxController.y().onFalse(intake.stop());
```

Comment:
Trigger functions return the trigger, so you can chain calls:
```suggestion
        xboxController.y().whileTrue(commandGroups.intake()).onFalse(intake.stop());
```